kfs-core/src/main/java/edu/arizona/kfs/vnd/service/impl/EcustomsServiceImpl.java

```diff
@@ -0,0 +1,698 @@
+    private static transient volatile EcustomsFileService ecustomsFileService;
+
+    private File dataFile;
+    private File vendorCountFile;
+
+    private String defaultLastRunDate;
+    private String batchFileName;
+    private String batchName;
+
+    public String getDefaultLastRunDate() {
+        return defaultLastRunDate;
+    }
+
+    public void setDefaultLastRunDate(String defaultLastRunDate) {
+        this.defaultLastRunDate = defaultLastRunDate;
+    }
+
+    public String getBatchFileName() {
+        return batchFileName;
+    }
+
+    public void setBatchFileName(String batchFileName) {
+        this.batchFileName = batchFileName;
+    }
+
+    private static VendorService getVendorService() {
+        if (vendorService == null) {
+            vendorService = SpringContext.getBean(VendorService.class);
+        }
+        return vendorService;
+    }
+
+    private static DocumentService getDocumentService() {
+        if (documentService == null) {
+            documentService = SpringContext.getBean(DocumentService.class);
+        }
+        return documentService;
+    }
+
+    private static BusinessObjectService getBusinessObjectService() {
+        if (businessObjectService == null) {
+            businessObjectService = SpringContext.getBean(BusinessObjectService.class);
+        }
+        return businessObjectService;
+    }
+
+    private static LookupService getLookupService() {
+        if (lookupService == null) {
+            lookupService = SpringContext.getBean(LookupService.class);
+        }
+        return lookupService;
+    }
+
+    private static EcustomsFileService getEcustomsFileService() {
+        if (ecustomsFileService == null) {
+            ecustomsFileService = SpringContext.getBean(EcustomsFileService.class);
+        }
+        return ecustomsFileService;
+    }
+
+    @Override
+    public boolean createEcustomsDailyFile(String jobName, Date jobRunDate) throws Exception {
+        LOG.debug("createEcustomsDailyFile(): jobRunDate=" + jobRunDate + ", start-time=" + new Date());
+        boolean processVendorDetails = false;
+        try {
+            boolean deleteOldDailyDone = getEcustomsFileService().deleteOldDailyDoneFiles();
+            LOG.debug("getEcustomsFileService().deleteOldDailyDoneFiles() returned " + deleteOldDailyDone);
+            dataFile = getEcustomsFileService().getDailyBatchDataFile(jobRunDate);
+            vendorCountFile = getEcustomsFileService().getDailyBatchVendorCountFile(jobRunDate);
+            batchName = VendorConstants.ECUSTOMS_BATCH_DAILY;
+            List<VendorDetail> vendorDetails = getDailyVendorDetailList(jobRunDate);
+            processVendorDetails = processVendorDetails(jobRunDate, vendorDetails);
+            getEcustomsFileService().createDailyBatchDoneFile(jobRunDate);
+        } catch (Exception e) {
+            LOG.error("Exception: " + e.getMessage());
+            throw new Exception(e.getMessage());

```

Comment:
An exception is being thrown either way, which will result in the job stopping, right? And it's fine that everything throws an E, it's that I just dont' like every method in the entire call stack logging the same exact error for no reason. If we aren't doing any exception handling, as in any work or changing the E message, which we aren't here at all in this example, then why log here at all? If you can come up with an argument why we would want to see the exact same error several times over in the log, then I'll let you keep this. As it stands now, I haven't heard a compelling reason yet. And if I am missing your point, please do let me know.